src/shared/components/VideoPreview/VideoPreview.tsx

```diff
@@ -104,10 +105,9 @@ const VideoPreview: React.FC<VideoPreviewProps> = ({
     if (!onClick) {
       return
     }
-    if (handleRemove) {
-      e.stopPropagation()
-      handleRemove()
-    }
+
+    e.stopPropagation()
+    updateWatchedVideos('REMOVED', id)
```

Comment:
That's not what I meant. Matter of fact, we cannot do that, this component is in `shared`, which means it cannot interact with `usePersonalData` which is Atlas-specific. This component should be purely presentational, it needs to handle all interaction via props and callback. Previous approach was ok, we just shouldn't pass callback via `Video` type - `handleRemove` should be a prop on `VideoGallery` as well